Tests failing after upstream change to LazyReferenceMapper

https://github.com/fsspec/kerchunk/blob/3c4e9fc960e159875e8f258ccd20fdbc565513df/kerchunk/df.py#L154-L156 relies on `record_size` being the first positional parameter in `LazyReferenceMapper.create`, but https://github.com/fsspec/filesystem_spec/pull/1420 changed the order such that `root` is first. This is leading to failing tests in Kerchunk (see below). The simple fix would be to update the argument order in `refs_to_dataframe`, but before opening a PR I wanted to check with @martindurant if this is preferable to maintaining backwards compatibility upstream?

```
FAILED kerchunk/tests/test_combine.py::test_lazy_filler - AttributeError: 'int' object has no attribute 'startswith'
FAILED kerchunk/tests/test_combine_concat.py::test_success[arrays0-chunks0-0] - TypeError: LazyReferenceMapper.create() got multiple values for argument 'root'
FAILED kerchunk/tests/test_combine_concat.py::test_success[arrays1-chunks1-0] - TypeError: LazyReferenceMapper.create() got multiple values for argument 'root'
FAILED kerchunk/tests/test_combine_concat.py::test_success[arrays2-chunks2-0] - TypeError: LazyReferenceMapper.create() got multiple values for argument 'root'
FAILED kerchunk/tests/test_combine_concat.py::test_success[arrays3-chunks3-1] - TypeError: LazyReferenceMapper.create() got multiple values for argument 'root'
FAILED kerchunk/tests/test_combine_concat.py::test_success[arrays4-chunks4-0] - TypeError: LazyReferenceMapper.create() got multiple values for argument 'root'
FAILED kerchunk/tests/test_df.py::test_1[True] - TypeError: LazyReferenceMapper.create() got multiple values for argument 'root'
FAILED kerchunk/tests/test_df.py::test_1[False] - TypeError: LazyReferenceMapper.create() got multiple values for argument 'root'
````

--- a/kerchunk/df.py
+++ b/kerchunk/df.py
@@ -152,7 +152,10 @@ def refs_to_dataframe(
 
     fs, _ = fsspec.core.url_to_fs(url, **(storage_options or {}))
     out = LazyReferenceMapper.create(
-        record_size, root=url, fs=fs, categorical_threshold=categorical_threshold
+        record_size=record_size,
+        root=url,
+        fs=fs,
+        categorical_threshold=categorical_threshold,
     )
 
     for k in sorted(refs):

--- a/kerchunk/tests/test_combine.py
+++ b/kerchunk/tests/test_combine.py
@@ -310,7 +310,7 @@ def test_lazy_filler(tmpdir, refs):
 
     fs = fsspec.filesystem("file")
     tmpdir = str(tmpdir)
-    out = LazyReferenceMapper.create(10, tmpdir, fs)
+    out = LazyReferenceMapper.create(record_size=10, root=tmpdir, fs=fs)
 
     mzz = MultiZarrToZarr(
         [refs["single1"], refs["single2"]],

